tests/functional/test_e2e.py

```diff
@@ -453,6 +453,66 @@ def test_e2e_release_files(
     expect(page.locator("body")).not_to_contain_text("test-workspace by researcher")
 
 
+def test_e2e_update_file(page, live_server, dev_users):
+    """
+    Test researcher updates a modified file in a returned request
+    """
+    # set up a returned file & request
+    author = factories.create_user("researcher", ["test-workspace"], False)
+    path = "subdir/file.txt"
+
+    factories.create_request_at_status(
+        "test-workspace",
+        author=author,
+        status=RequestStatus.RETURNED,
+        files=[factories.request_file(path=path, group="default", rejected=True)],
+    )
+
+    # change the file on disk
+    workspace = bll.get_workspace("test-workspace", author)
+    factories.write_workspace_file(workspace, path, contents="changed")
+
+    # Log in as researcher
+    login_as(live_server, page, "researcher")
+
+    # Click on to workspaces link
+    find_and_click(page.get_by_test_id("nav-workspaces"))
+    expect(page.locator("body")).to_contain_text("Workspaces for researcher")
+
+    # Click on the workspace
+    find_and_click(page.locator("#workspaces").get_by_role("link"))
+    expect(page.locator("body")).to_contain_text("subdir")
+    # subdirectories start off collapsed; the file links are not present
+    assert page.get_by_role("link", name="file.txt").all() == []
+    assert page.get_by_role("link", name="file.foo").all() == []
+
+    # Click on the subdir and then the file link to view in the workspace
+    # There will be more than one link to the folder/file in the page,
+    # one in the explorer, and one in the main folder view
+    # Click on the first link we find
+    find_and_click(page.get_by_role("link", name="subdir").first)
+
```

Comment:
Everything from logging in to here is already tested in other functional tests, I think. You could skip it and go straight to the directory page with `page.goto(workspace.get_url("subdir")`

Reply:
good shout, thanks, done